addons/mail/static/src/model/model_field.js

```diff
@@ -295,18 +295,15 @@ class ModelField {
             const OtherModel = otherRecord.constructor;
             const otherField = OtherModel.__fieldMap[relatedFieldName];
             const newVal = otherField.get(otherRecord);
+            if (newVal === undefined) {
+                return clear();
+            }
             if (this.fieldType === 'relation') {
-                if (newVal) {
-                    return [['replace', newVal]];
-                } else {
-                    return [['unlink-all']];
-                }
+                return [['replace', newVal]];
```

Comment:
could use `replace` method

Reply:
At forward-port, it doesn't exist in v14.